metrics-operator/controllers/common/providers/datadog/datadog.go

```diff
@@ -16,6 +16,10 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+var errNoValues = "No values in query result"
+var errNoMetricPoints = "No metric points in query result"
+var errCloseBody = "Could not close request body"
```

Comment:
Let's start those error messages with a lower case character so we stick with go best practices:

```suggestion
var errNoValues = "no values in query result"
var errNoMetricPoints = "no metric points in query result"
var errCloseBody = "could not close request body"
```

Reply:
a link with some documentation about standard error messages in Golang: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
these could be directly errors with `errors.New()` so you don't need a call to `fmt.Errorf` each time and they could be used with `errors.Is`

Reply:
fixed